File: 2654-minimum-number-of-operations-to-make-all-array-elements-equal-to-1/2654-minimum-number-of-operations-to-make-all-array-elements-equal-to-1.py
class Solution:
    def minOperations(self, nums: List[int]) -> int:
        n = len(nums)
        total_gcd = reduce(gcd, nums)
        if total_gcd != 1:
            return -1
        
        # 若已有 1
        ones = nums.count(1)
        if ones > 0:
            return n - ones
        
        # 沒有 1：找最短 gcd = 1 的子陣列
        min_len = float('inf')
        for i in range(n):
            cur_gcd = nums[i]
            for j in range(i + 1, n):
                cur_gcd = gcd(cur_gcd, nums[j])
                if cur_gcd == 1:
                    min_len = min(min_len, j - i + 1)
                    break # 不需繼續擴展此區段
        
        return (min_len - 1) + (n - 1)

File: 2654-minimum-number-of-operations-to-make-all-array-elements-equal-to-1/2654-minimum-number-of-operations-to-make-all-array-elements-equal-to-1.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution {
public:
    int minOperations(vector<int>& nums) {
        int n = nums.size();

        // Step 1: 若整體 gcd > 1，無解
        int total_gcd = nums[0];
        for (int i = 1; i < n; ++i) {
            total_gcd = gcd(total_gcd, nums[i]);
        }
        if (total_gcd != 1)
            return -1;

        // Step 2: 若已有 1，直接計算非 1 元素個數
        int ones = count(nums.begin(), nums.end(), 1);
        if (ones > 0)
            return n - ones;

        // Step 3: 沒有 1 → 找最短 gcd = 1 的子陣列
        int minLen = INT_MAX;
        for (int i = 0; i < n; ++i) {
            int cur_gcd = nums[i];
            for (int j = i + 1; j < n; ++j) {
                cur_gcd = gcd(cur_gcd, nums[j]);
                if (cur_gcd == 1) {
                    minLen = min(minLen, j - i + 1);
                    break;  // 再延伸也不會更短，直接跳出
                }
            }
        }

        // Step 4: 總操作數 = (L - 1) + (n - 1)
        return (minLen - 1) + (n - 1);
    }
};
